WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -1926,4 +1926,11 @@ public static void openDomainManagement(@NonNull Context context) {
         Intent intent = new Intent(context, DomainManagementActivity.class);
         context.startActivity(intent);
     }
+
+    public static void openShareIntent(@NonNull Context context, @NonNull String link) {
+        Intent intent = new Intent(Intent.ACTION_SEND);
+        intent.setType("text/plain");
+        intent.putExtra(Intent.EXTRA_TEXT, link);
```

Comment:
Hey @zwarm 
I added a line here to show the title in the share sheet. I think seeing that title helps the user in making sure that the right link is being shared(Saw this tip in some article). In case, the title is null, then the share sheet is as you have attached in the PR description. 

<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/c3bdb927-a161-43b1-abef-e1fb069d8935" width="200" height="480">

The commit for the change is here 02ce0d5f65593193636c832f98f6366416bad52b. In case, you think this is unnecessary, I can revert it from the feature branch after merge